### PR TITLE
FE-2152 - [Radio Button] - should discard the name property

### PR DESCRIPTION
**Pull request by dawidzarzycki**
# Description
- Removed name prop from RadioButton. The prop has been replaced by groupName provided by RadioButtonGroup.

### CODE REVIEW

File: src/__experimental__/components/radio-button/radio-button-group.component.js [hunk 1]
```diff
@@ -34,7 +36,8 @@ const RadioButtonGroup = (props) => {
         checked,
         inputName: groupName,
         onChange: handleChange,
-        tabindex
+        tabindex,
+        groupName
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.component.js]: We don't need to pass down the `groupName` prop, its already passed as `inputName`

File: src/__experimental__/components/radio-button/radio-button.component.js [hunk 2]
```diff
@@ -82,7 +82,15 @@ RadioButton.propTypes = {
    * The name of the group containing the RadioButton (can also be set via
    * the 'groupName' prop of the RadioButtonGroup component)
    */
```
Comment (mkrds) [inline src/__experimental__/components/radio-button/radio-button.component.js]: Not sure if that's enough, you can still pass the name prop to the RadioButton which gets passed on to CheckableInput even though it warns the user about not doing so

Comment (dawidzarzycki) [reply src/__experimental__/components/radio-button/radio-button.component.js]: Sure, you can pass the prop but you will get the console error that you shouldn't do it. The same situation is in other libraries

File: src/__experimental__/components/radio-button/radio-button-group.component.js (hunk 1 above)
Comment (dawidzarzycki) [reply src/__experimental__/components/radio-button/radio-button-group.component.js]: Oh... how could I missed that one

File: src/__experimental__/components/radio-button/radio-button.component.js (hunk 2 above)
Comment (jamime) [reply src/__experimental__/components/radio-button/radio-button.component.js]: I agree with @mkrds we should discard the property AND warn about it.

Comment (dawidzarzycki) [reply src/__experimental__/components/radio-button/radio-button.component.js]: I'm not big fan of the solution but I lost 2:1 :D 
Updated in https://github.com/Sage/carbon/pull/2389/commits/41792bfa5183738d2d45bf3ad66fe318edb8df1a

Comment (jamime) [reply src/__experimental__/components/radio-button/radio-button.component.js]: Yeah, delete is very slow. Can you not refactor it so we don't have three names for the same thing (`name`, `groupName`, `inputName`). Really we need to rename the `inputName` prop to be `name`?

Comment (mkrds) [reply src/__experimental__/components/radio-button/radio-button.component.js]: Not a fan of that solution either, but I guess that thanks to that we can free users of some hard to debug bugs. Anyway I'd rather not mutate the prop object and insted of deleting one property I'd just desctructure it like that:
`
const RadioButton = ({
  id, label, onChange, value, name, ...props
})
`

Comment (jamime) [reply src/__experimental__/components/radio-button/radio-button.component.js]: @mkrds that fails eslint rule, `no-unused-variables`. Why do you not want to rename `inputName`?
The intended interface is `<RadioGroup name="foobar"><RadioButton /></RadioGroup>`

File: change_log/next/FE-2152-radio-button.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+New Features: "Removed name prop from RadioButton. The prop has been replaced by groupName provided by RadioButtonGroup. (Component: RadioButton, RadioButtonGroup)"
```
Comment (jamime) [inline change_log/next/FE-2152-radio-button.yml]:
```suggestion
Breaking Changes: "Renamed `RadioButtonGroup` prop `groupName` to `name`(Component: RadioButtonGroup)"
```

File: src/__experimental__/components/radio-button/radio-button-group.component.js [hunk 4]
```diff
@@ -47,9 +46,10 @@ RadioButtonGroup.propTypes = {
   /** The RadioButton objects to be rendered in the group */
   children: PropTypes.node.isRequired,
   /** Specifies the name prop to be applied to each button in the group */
-  groupName: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
   /** The content for the RadioGroup Legend */
   legend: PropTypes.string.isRequired
+
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.component.js]:
```suggestion

```

**Comment by jamime** [comment]
@edleeks87 It's because that branch that you used still used `groupName`. I've checked and I am happy with these changes. @dawidzarzycki please can you squash all of these changes into a single commit and rebase off master.

**Comment by jamime** [comment]
I squashed the changes and assigned to @grabkowski to merge after https://github.com/Sage/carbon/pull/2269 has been merged.

**Comment by dawidzarzycki** [comment]
> I squashed the changes and assigned to @grabkowski to merge after #2269 has been merged.

The squash looks good to me

**Comment by jamime** [comment]
Closing this is now included in PR https://github.com/Sage/carbon/pull/2269

**Pull request closed without merging**